### DISCUSSION TITLE
How do you menage such a clean changelog without a systematic approach e.g. SemVer

### DISCUSSION THREAD

**Discussion by maxdd** [dani-garcia/vaultwarden#3507] [General]
Hello @dani-garcia,
this is not strictly related to vaultwarden product but more to the code lifecycle in this repo.
I'm curious to know whether the content of each release https://github.com/dani-garcia/vaultwarden/releases is hand-written or whether you are using a tool which infers the version/commits/authors automatically like a sort of semantic versioning.
It is indeed very clean and if it is done by hand i can only imagine the effort behind.

Regards,

**Comment by BlackDex** [comment]
@maxdd, when using PR's for all your changes, you can use the GitHub CLI, `gh` after you created a tagged version.

An example would be.

```bash
gh release create v1.28.1 --generate-notes
```
